error: lint `clippy::safe_fn` has been renamed to `clippy::safe_code`
   |
   = help: use the new name `clippy::safe_code`
   = note: requested on the command line with `-D clippy::safe_fn`
   = note: `-D renamed-and-removed-lints` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(renamed_and_removed_lints)]`

error: lint `clippy::safe_fn` has been renamed to `clippy::safe_code`
   |
   = help: use the new name `clippy::safe_code`
   = note: requested on the command line with `-D clippy::safe_fn`
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: item must be `pub`
  --> tests/ui/map_err.rs:7:1
   |
LL | enum Errors {
   | ^ help: make this item public: `pub`
   |
   = note: requested on the command line with `-D clippy::missing-pub`

error: item must be `pub`
  --> tests/ui/map_err.rs:6:10
   |
LL | #[derive(Debug)]
   |          ^

error: function must be `unsafe`
  --> tests/ui/map_err.rs:6:7
   |
LL | #[derive(Debug)]
   |       ^  ----- in this derive macro expansion
   |
help: make this function unsafe: `unsafe`
  --> tests/ui/map_err.rs:6:7
   |
LL | #[derive(Debug)]
   |       ^  ----- in this derive macro expansion
   = note: `-D clippy::safe-code` implied by `-D clippy::safe-fn`
   = help: to override `-D clippy::safe-fn` add `#[allow(clippy::safe_code)]`

error: parameter must be `mut`
  --> tests/ui/map_err.rs:6:10
   |
LL | #[derive(Debug)]
   |          ^ help: make this parameter mutable: `mut`
   |
   = note: requested on the command line with `-D clippy::missing-mut`

error: parameter must be `mut`
  --> tests/ui/map_err.rs:6:10
   |
LL | #[derive(Debug)]
   |          ^

error: references are not allowed
  --> tests/ui/map_err.rs:6:10
   |
LL | #[derive(Debug)]
   |          ^^^^^
   |
   = help: use a raw pointer instead: `*const`
   = note: requested on the command line with `-D clippy::reference-used`

error: references are not allowed
  --> tests/ui/map_err.rs:6:10
   |
LL | #[derive(Debug)]
   |          ^^^^^
   |
   = help: use a raw pointer instead: `*raw`

error: item must be `pub`
  --> tests/ui/map_err.rs:11:1
   |
LL | impl Error for Errors {}
   | ^ help: make this item public: `pub`

error: item must be `pub`
  --> tests/ui/map_err.rs:13:1
   |
LL | impl fmt::Display for Errors {
   | ^ help: make this item public: `pub`

error: function must be `unsafe`
  --> tests/ui/map_err.rs:14:5
   |
LL |     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |     ^
   |
help: make this function unsafe: `unsafe`
  --> tests/ui/map_err.rs:14:5
   |
LL |     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |     ^

error: parameter must be `mut`
  --> tests/ui/map_err.rs:14:13
   |
LL |     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |             ^ help: make this parameter mutable: `mut`

error: parameter must be `mut`
  --> tests/ui/map_err.rs:14:19
   |
LL |     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |                   ^ help: make this parameter mutable: `mut`

error: references are not allowed
  --> tests/ui/map_err.rs:14:12
   |
LL |     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |            ^^^^^
   |
   = help: use a raw pointer instead: `*const`

error: references are not allowed
  --> tests/ui/map_err.rs:14:22
   |
LL |     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |                      ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use a raw pointer instead: `*raw`

error: item must be `pub`
  --> tests/ui/map_err.rs:19:1
   |
LL | fn main() -> Result<(), Errors> {
   | ^ help: make this item public: `pub`

error: function must be `unsafe`
  --> tests/ui/map_err.rs:19:1
   |
LL | fn main() -> Result<(), Errors> {
   | ^
   |
help: make this function unsafe: `unsafe`
  --> tests/ui/map_err.rs:19:1
   |
LL | fn main() -> Result<(), Errors> {
   | ^

error: variable must be `mut`
  --> tests/ui/map_err.rs:20:9
   |
LL |     let x = u32::try_from(-123_i32);
   |         ^ help: make this variable mutable: `mut`

error: lint `clippy::safe_fn` has been renamed to `clippy::safe_code`
   |
   = help: use the new name `clippy::safe_code`
   = note: requested on the command line with `-D clippy::safe_fn`
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: missing `#![no_std]`
   |
   = help: make this crate `#![no_std]`
   = note: requested on the command line with `-D clippy::missing-no-std`

error: aborting due to 21 previous errors

